profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.multilingual.inc

```diff
@@ -0,0 +1,701 @@
+    $tamper += array(
+      'settings' => array(
+        'entity_type' => 'taxonomy_term',
+        'bundle' => $settings['vocabulary'],
+        'column' => 'value',
+        'property' => FALSE,
+        'update' => 'Update',
+        'field' => $settings['field'],
+      ),
+      'description' => 'Entity Field Query finder',
+    );
+  }
+
+  // Prepare setting for Rewrite plugin.
+  if ($settings['plugin_id'] == 'rewrite') {
+    $tamper += array(
+      'settings' => array(
+        'text' => $settings['prefix'] . '[xpathparser:' . $i . ']',
+      ),
+      'description' => 'Rewrite',
+    );
+  }
+
+  // String to timestamp plugin.
+  if ($settings['plugin_id'] == 'strtotime') {
+    $tamper += array(
+      'settings' => '',
+      'description' => 'String to Unix timestamp',
+    );
+  }
+
+  return $tamper;
+}
+
+/**
+ * Build tampers programmatically.
+ */
+function _nexteuropa_newsroom_dynamic_tamper($tamper) {
+  $def_lang = language_default()->language;
+  $feeds_tamper = new stdClass();
+  $feeds_tamper->disabled = FALSE;
+  $feeds_tamper->api_version = 2;
+  $feeds_tamper->weight = 0;
+  if ($tamper == NULL) {
+    $feeds_tamper->id = 'newsroom_items_multilingual-blank_source_1-default_value';
+    $feeds_tamper->importer = NEXTEUROPA_NEWSROOM_ITEM_IMPORTER;
+    $feeds_tamper->source = 'Blank source 1';
+    $feeds_tamper->plugin_id = 'default_value';
+    $feeds_tamper->settings = array(
+      'default_value' => $def_lang,
+    );
+    $feeds_tamper->description = 'Set default language';
+  }
+  else {
+    $feeds_tamper->id = $tamper['id'];
+    $feeds_tamper->importer = $tamper['importer'];
+    $feeds_tamper->source = $tamper['source'];
+    $feeds_tamper->plugin_id = $tamper['plugin_id'];
+    $feeds_tamper->settings = $tamper['settings'];
+    $feeds_tamper->description = $tamper['description'];
+  }
+
+  // Delete a pre-existing tamper with the same id.
+  db_delete('feeds_tamper')
+    ->condition('id', $feeds_tamper->id)
+    ->execute();
+  // Save the tamper.
+  feeds_tamper_save_instance($feeds_tamper);
+}
+
+/**
+ * Helper function to get the enabled languages.
+ */
+function _nexteuropa_newsroom_languages() {
+  $lang = locale_language_list('name');
+  $lang_keys = array_keys($lang);
+  $eng = array_search('en', $lang_keys);
+  if ($eng) {
+    unset($lang_keys[$eng]);
+    array_unshift($lang_keys, 'en');
+  }
+  return $lang_keys;
+}
+
+/**
+ * Check if a field is translatable.
+ *
+ * @param array $premapping
+ *   Premappings array.
+ */
+function _nexteuropa_newsroom_check_translations(array $premapping) {
+  // Build an array with langauge settings for the fields.
+  foreach ($premapping as $fieldname => $values) {
+    // Sanitize fieldname.
+    $attribute = strrchr($fieldname, ':');
+    $sanitized_name = isset($values['original_field']) ? $values['original_field'] : str_replace($attribute, '', $fieldname);
+    $field = field_info_field($sanitized_name);
```

Comment:
always asking field_info_field makes the function quite heavy, try to refactor it and use field_info_fields what gives you all of the fields in one big array in one step where you can check all of the translatable availabilities. Maybe in this case you also can put the translatable fields in a variable and ask it when you need to work with them, because these are very static data, no need to ask it always (cache solution is also not the best idea here).